### PR TITLE
[1.0.5] Update Heltec ESP32 series boards definition

### PR DESCRIPTION
**Pull request by Heltec-Aaron-Lee**
- Added LoRaWAN features option for Heltec ESP32 + LoRa series boards;
- Added a new board Heltec Wireless Stick Lite.

### CODE REVIEW

File: platform.txt [hunk 1]
```diff
@@ -22,7 +22,7 @@ compiler.warning_flags.all=-Wall -Werror=all -Wextra
...
...mpiler.sdk.path}/include/mdns" "-I{compiler.sdk.path}/include/micro-ecc" "-I{compiler.sdk.path}/include/mqtt" "-I{compiler.sdk.path}/include/newlib" "-I{compiler.sdk.path}/include/nghttp" "-I{compiler.sdk.path}/include/nimble" "-I{compiler.sdk.path}/include/nvs_flash" "-I{compiler.sdk.path}/include/openssl" "-I{compiler.sdk.path}/include/protobuf-c" "-I{compiler.sdk.path}/include/protocomm" "-I{compiler.sdk.path}/include/pthread" "-I{compiler.sdk.path}/include/sdmmc" "-I{compiler.sdk.path}/include/smartconfig_ack" "-I{compiler.sdk.path}/include/soc" "-I{compiler.sdk.path}/include/spi_flash" "-I{compiler.sdk.path}/include/spiffs" "-I{compiler.sdk.path}/include/tcp_transport" "-I{compiler.sdk.path}/include/tcpip_adapter" "-I{compiler.sdk.path}/include/ulp" "-I{compiler.sdk.path}/include/unity" "-I{compiler.sdk.path}/include/vfs" "-I{compiler.sdk.path}/include/wear_levelling" "-I{compiler.sdk.path}/include/wifi_provisioning" "-I{compiler.sdk.path}/include/wpa_supplicant" "-I{compiler.sdk.path}/include/xtensa-debug-module" "-I{compiler.sdk.path}/include/esp-face" "-I{compiler.sdk.path}/include/esp32-camera" "-I{compiler.sdk.path}/include/esp-face" "-I{compiler.sdk.path}/include/fb_gfx"
```
Comment (me-no-dev) [inline platform.txt:25]: these changes will break it for every other board :) and this is not the correct place for such changes as this line get's replaced automatically by the lib builder

File: platform.txt [hunk 2]
```diff
@@ -51,6 +51,8 @@ build.flash_size=4MB
 build.flash_mode=dio
 build.boot=bootloader
 build.code_debug=0
+build.band=LORAMAC_REGION_EU868
+build.LoRaWanDebugLevel=0
 build.defines=
 build.extra_flags=-DESP32 -DCORE_DEBUG_LEVEL={build.code_debug} {build.defines}
```
Comment (me-no-dev) [inline platform.txt:57]: You should use [board].build.defines to add the necessary flags

**Comment by Heltec-Aaron-Lee** [comment]
Hi @me-no-dev 

Thank you for your hints, I had fixed the PR request, please confirm again.

File: variants/heltec_wifi_kit_32/pins_arduino.h [hunk 3]
```diff
@@ -17,7 +17,6 @@
 
 static const uint8_t LED_BUILTIN = 25;
 #define BUILTIN_LED  LED_BUILTIN // backward compatibility
-#define LED_BUILTIN LED_BUILTIN
```
Comment (me-no-dev) [inline variants/heltec_wifi_kit_32/pins_arduino.h:20]: do not remove those lines ;) they are there for a reason

File: variants/heltec_wifi_lora_32/pins_arduino.h [hunk 4]
```diff
@@ -17,7 +17,6 @@
 
 static const uint8_t LED_BUILTIN = 25;
 #define BUILTIN_LED  LED_BUILTIN // backward compatibility
-#define LED_BUILTIN LED_BUILTIN
```
Comment (me-no-dev) [inline variants/heltec_wifi_lora_32/pins_arduino.h:20]: same here

File: variants/heltec_wifi_lora_32_V2/pins_arduino.h [hunk 5]
```diff
@@ -17,7 +17,6 @@
 
 static const uint8_t LED_BUILTIN = 25;
 #define BUILTIN_LED  LED_BUILTIN // backward compatibility
-#define LED_BUILTIN LED_BUILTIN
```
Comment (me-no-dev) [inline variants/heltec_wifi_lora_32_V2/pins_arduino.h:20]: and here

File: variants/heltec_wireless_stick/pins_arduino.h [hunk 6]
```diff
@@ -17,7 +17,6 @@
 
 static const uint8_t LED_BUILTIN = 25;
 #define BUILTIN_LED  LED_BUILTIN // backward compatibility
-#define LED_BUILTIN LED_BUILTIN
```
Comment (me-no-dev) [inline variants/heltec_wireless_stick/pins_arduino.h:20]: and here

File: variants/heltec_wireless_stick_lite/pins_arduino.h [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+#define EXTERNAL_NUM_INTERRUPTS 16
+#define NUM_DIGITAL_PINS        40
+#define NUM_ANALOG_INPUTS       16
+
+#define analogInputToDigitalPin(p)  (((p)<20)?(esp32_adc2gpio[(p)]):-1)
+#define digitalPinToInterrupt(p)    (((p)<40)?(p):-1)
+#define digitalPinHasPWM(p)         (p < 34)
+
+static const uint8_t LED_BUILTIN = 25;
+#define BUILTIN_LED  LED_BUILTIN // backward compatibility
```
Comment (me-no-dev) [inline variants/heltec_wireless_stick_lite/pins_arduino.h:19]: add `#define LED_BUILTIN LED_BUILTIN`

File: variants/heltec_wifi_kit_32/pins_arduino.h (hunk 3 above)
Comment (me-no-dev) [reply variants/heltec_wifi_kit_32/pins_arduino.h:20]: This is so things like `#ifdef LED_BUILTIN` can work :)

**Comment by me-no-dev** [comment]
Almost there :) sorry I missed the deleted definitions earlier

**Comment by Heltec-Aaron-Lee** [comment]
Hi @me-no-dev 

Thank you for your fast response, just updated. 😃

**Comment by me-no-dev** [comment]
@Heltec-Aaron-Lee waiting for CI to complete and will merge :)

**Comment by Heltec-Aaron-Lee** [comment]
👍 👍 👍

**Pull request merged**